kie-api/src/main/java/org/kie/api/builder/ReleaseId.java

```diff
@@ -20,6 +20,14 @@
  * ReleaseId is a full identifier far a given version of an artifact.
  * Following the Maven conventions it is composed of 3 parts: a groupId, an artifactId and a version
  */
-public interface ReleaseId extends org.appformer.maven.support.AFReleaseId {
+public interface ReleaseId extends org.appformer.maven.support.AFReleaseId, Comparable<ReleaseId> {
+
+    @Override
+    default int compareTo(ReleaseId that) {
+        if (this == that) {
+            return 0;
+        }
+        return new ReleaseIdComparator().compare(this, that);
```

Comment:
@errantepiphany I don't like to have that ReleaseIdComparator class among our public API. I believe it should be moved inside drools. It's ok for the ReleaseId interface to extend Comparable, but I don't see the need for this default method. I think that the default method should be removed and the implementation of compareTo added directly inside the ReleaseIdImpl class.